doctr/utils/geometry.py

```diff
@@ -145,22 +145,62 @@ def rotate_abs_boxes(boxes: np.ndarray, angle: float, img_shape: Tuple[int, int]
     return rotated_boxes
 
 
+def remap_boxes(boxes: np.ndarray, orig_shape: Tuple[int, int], dest_shape: Tuple[int, int]) -> np.ndarray:
+    """ Remaps a batch of RotatedBbox (x, y, w, h, alpha) expressed for an origin_shape to a destination_shape.
+    This does not impact the absolute shape of the boxes, but allow to calculate the new relative RotatedBbox
+    coordinates after a resizing of the image.
+
+    Args:
+        boxes: (N, 5) array of RELATIVE RotatedBbox (x, y, w, h, alpha)
+        orig_shape: shape of the origin image
+        dest_shape: shape of the destination image
+
+    Returns:
+        A batch of rotated boxes (N, 5): (x, y, w, h, alpha) expressed in the destination referencial
+
+    """
+
+    if len(dest_shape) != 2:
+        raise ValueError(f"Mask length should be 2, was found at: {len(dest_shape)}")
+    if len(orig_shape) != 2:
+        raise ValueError(f"Image_shape length should be 2, was found at: {len(orig_shape)}")
+    orig_height, orig_width = orig_shape
+    dest_height, dest_width = dest_shape
+    mboxes = boxes.copy()
+    mboxes[:, 0] = ((boxes[:, 0] * orig_width) + (dest_width - orig_width) / 2) / dest_width
+    mboxes[:, 1] = ((boxes[:, 1] * orig_height) + (dest_height - orig_height) / 2) / dest_height
+    mboxes[:, 2] = boxes[:, 2] * orig_width / dest_width
+    mboxes[:, 3] = boxes[:, 3] * orig_height / dest_height
+    return mboxes
+
+
 def rotate_boxes(
     boxes: np.ndarray,
     angle: float = 0.,
-    min_angle: float = 1.
+    min_angle: float = 1.,
+    orig_shape: Optional[Tuple[int, int]] = None,
+    mask_shape: Optional[Tuple[int, int]] = None,
 ) -> np.ndarray:
-    """Rotate a batch of straight bounding boxes (xmin, ymin, xmax, ymax) of an angle,
-    if angle > min_angle, around the center of the page.
+    """Rotate a batch of straight bounding boxes (xmin, ymin, xmax, ymax) or rotated bounding boxes
+    (x, y, w, h, alpha) of an angle, if angle > min_angle, around the center of the page.
 
     Args:
-        boxes: (N, 4) array of RELATIVE boxes
+        boxes: (N, 4) or (N, 5) array of RELATIVE boxes
         angle: angle between -90 and +90 degrees
         min_angle: minimum angle to rotate boxes
+        orig_shape: shape of the origin image
+        mask_shape: shape of the mask if the image is cropped after the rotation
 
     Returns:
         A batch of rotated boxes (N, 5): (x, y, w, h, alpha) or a batch of straight bounding boxes
     """
+    # Change format of the boxes to rotated boxes
+    if boxes.shape[1] == 4:
+        boxes = np.column_stack(((boxes[:, 0] + boxes[:, 2]) / 2,
+                                 (boxes[:, 1] + boxes[:, 3]) / 2,
+                                 boxes[:, 2] - boxes[:, 0],
+                                 boxes[:, 3] - boxes[:, 1],
+                                 np.zeros(boxes.shape[0])))
```

Comment:
I agree with the conversion but I would argue it's better to avoid inplace modifications of the input arg

Reply:
oh yes, sorry, i changed this